rotkehlchen/chain/ethereum/modules/yearn/decoder.py

```diff
@@ -92,102 +110,264 @@ def reload_data(self) -> None:
-            context.event.counterparty = protocol
-            vault_token_name = _get_vault_token_name(context.transaction.to_address)
-            context.event.notes = f'Deposit {context.event.balance.amount} {context.token.symbol} in {protocol} vault {vault_token_name}'  # noqa: E501
-        elif (
-            is_deposit is True and
-            context.event.event_type == HistoryEventType.RECEIVE and
-            context.event.address == ZERO_ADDRESS
-        ):
-            context.event.event_subtype = HistoryEventSubType.RECEIVE_WRAPPED
-            context.event.counterparty = protocol
-            vault_token_name = _get_vault_token_name(context.transaction.to_address)
-            context.event.notes = f'Receive {context.event.balance.amount} {vault_token_name} after deposit in a {protocol} vault'  # noqa: E501
-        elif (
-            is_deposit is False and
-            context.event.event_type == HistoryEventType.RECEIVE and
-            context.event.address == context.transaction.to_address
-        ):
-            context.event.event_type = HistoryEventType.WITHDRAWAL
-            context.event.event_subtype = HistoryEventSubType.REMOVE_ASSET
-            context.event.counterparty = protocol
-            vault_token_name = _get_vault_token_name(context.transaction.to_address)
-            context.event.notes = f'Withdraw {context.event.balance.amount} {context.token.symbol} from {protocol} vault {vault_token_name}'  # noqa: E501
-        elif (
-            is_deposit is False and
-            context.event.event_type == HistoryEventType.SPEND and
-            context.event.address == ZERO_ADDRESS
-        ):
-            context.event.event_subtype = HistoryEventSubType.RETURN_WRAPPED
-            context.event.counterparty = protocol
-            context.event.notes = f'Return {context.event.balance.amount} {context.token.symbol} to a {protocol} vault'  # noqa: E501
+        if (vault_address := context.transaction.to_address) is None:
+            return None
+
+        event_token = self.base.get_or_create_evm_token(context.tx_log.address)
+        amount = token_normalized_value(
+            token_amount=int.from_bytes(context.tx_log.data[0:32]),
+            token=event_token,
+        )
+
+        if context.transaction.to_address in self.vaults[CPT_YEARN_V1]:
+            return CPT_YEARN_V1, event_token, amount, vault_address
+        elif context.transaction.to_address in self.vaults[CPT_YEARN_V2]:
+            return CPT_YEARN_V2, event_token, amount, vault_address
+
+        return None
+
+    def _maybe_get_other_transfer_event(
+            self,
+            context: DecoderContext,
+            event_type: HistoryEventType,
+            event_subtype: HistoryEventSubType,
+    ) -> 'EvmEvent | None':
+        """"""
+        for event in context.decoded_events:
+            if event.event_type == event_type and event.event_subtype == event_subtype:
```

Comment:
Isn't this too wide a matching? Just type/subtype? Shouldn't you narrow it down further?

Reply:
I added a check for counterparty, which seems to be pretty much the only other thing that can be checked for easily, since asset and amount for the event it's matching are unknown at the point where this is being used. I think checking counterparty and event type/subtype should be sufficient.

Reply:
Also added check for `location_label`, which should always match `context.transaction.from_address`

Reply:
There simply does not seem to be enough data available in the log to match this any better.
I've removed the `location_label` check again and put a note in the docstring as discussed in [discord](https://discord.com/channels/657906918408585217/1080783409640648765/1298994891011133521)